File: Tools/WinLicenseV2.0.8.0/WinlicenseSDK/ExamplesSDK/Text Keys Registration/C/CBuilder/Unit1.cpp
//---------------------------------------------------------------------------

#include <vcl.h>
#pragma hdrstop

#include "Unit1.h"
#include "..\..\..\..\Include\C\WinlicenseSDK.h"
#pragma comment (lib,"..\\..\\..\\..\\Lib\\OMF\\WinlicenseSDK.lib")

//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma resource "*.dfm"
TForm1 *Form1;
//---------------------------------------------------------------------------
__fastcall TForm1::TForm1(TComponent* Owner)
    : TForm(Owner)
{
    char    Name[200];
    char    Company[200];
    char    ExtraData[200];

    // check if application has been registered

    if (WLRegGetStatus(NULL) == 1)
    {
        Panel1->Visible = True;
        WLRegGetLicenseInfo(Name, Company, ExtraData);
        NameEdit->Text = Name;
        CompanyEdit->Text = Company;
        ExtradataEdit->Text = ExtraData;
    }

}
//---------------------------------------------------------------------------

void __fastcall TForm1::Button1Click(TObject *Sender)
{
    if (!WLRegNormalKeyCheck(TextKeyEdit->Text.c_str()))
    {
        MessageBox(NULL, "The key that you entered is invalid.\nPlease, try it again", "Error", MB_OK | MB_ICONERROR);
    }
    else
    {
        // if key is valid, insert the text key in a file
        // So, winlicense will be able to finish the registration process

        WLRegNormalKeyInstallToFile(TextKeyEdit->Text.c_str());
        MessageBox(NULL, "Registration key is correct. Please, restart\n this application to finish the registration process", "Success", MB_OK);
    }

}
//---------------------------------------------------------------------------

File: Tools/WinLicenseV2.0.8.0/WinlicenseSDK/ExamplesSDK/SmartActivate Keys/Delphi/Unit1.pas
unit Unit1;

interface

uses
  Windows, Messages, SysUtils, Variants, Classes, Graphics, Controls, Forms,
  Dialogs, StdCtrls, WinlicenseSDK;

type
  TForm1 = class(TForm)
    Label1: TLabel;
    Label2: TLabel;
    Button1: TButton;
    GroupBox1: TGroupBox;
    Label4: TLabel;
    Label5: TLabel;
    Label6: TLabel;
    Label3: TLabel;
    NameEdit: TEdit;
    OrganizationEdit: TEdit;
    ExtraDataEdit: TEdit;
    ActivationCodeEdit: TEdit;
    procedure Button1Click(Sender: TObject);
    procedure FormCreate(Sender: TObject);
  private
    { Private declarations }
  public
    { Public declarations }
  end;

var
  Form1: TForm1;

implementation

{$R *.dfm}

procedure TForm1.Button1Click(Sender: TObject);
begin

    if WLRegSmartKeyCheck(PChar(NameEdit.Text), PChar(OrganizationEdit.Text), PChar(ExtraDataEdit.Text), PChar(ActivationCodeEdit.Text)) = false  then

        MessageBox(0, 'Activation code is INVALID', 'Error', MB_OK or MB_ICONERROR)

    else
    begin

      // if activation is valid, insert the registration info in a file
      // So, winlicense will be able to finish the registration process

      WLRegSmartKeyInstallToFile(PChar(NameEdit.Text), PChar(OrganizationEdit.Text), PChar(ExtraDataEdit.Text), PChar(ActivationCodeEdit.Text));

      MessageBox(0, 'Activation code is correct. Please, restart this application to finish the registration process', 'Success', MB_OK);

    end;

end;

procedure TForm1.FormCreate(Sender: TObject);
var
   ExtStatus: Integer;
   
begin

   // check if application has been activated (registered)

    if WLRegGetStatus(ExtStatus) = 1 then
    begin

        Label1.Caption := 'This application has been ACTIVATED';
        Label2.Caption := '';
        Button1.Visible := False;
        GroupBox1.Visible := False;
    end;


end;

end.
